### ISSUE TITLE
Replace `@import` by `@use`?

### ISSUE THREAD

**Issue by robsonsobral** [HugoGiraudel/sass-guidelines#459]
I don't even understand `@use` yet, but they [say](https://sass-lang.com/blog/the-module-system-is-launched) it's the future.

**Comment by HugoGiraudel** [comment]
I guess at some point I should probably catch up with Sass and try to understand what’s new. 😅

**Comment by robsonsobral** [comment]
I came here to open an issue and... I already did!

---

I'm testing things. If I came to a conclusion, I come back and tell you!

**Comment by KittyGiraudel** [comment]
Alright! 😊

**Comment by robsonsobral** [comment]
Hey, big boss!

Remembering the current recommendation:

- `abstracts/`:
  - `_font-stacks.scss`
  - `_mixins.scss`
  - ...
- `base/`
- `components/`
- `layout/`
- `pages/`
- `themes/`
- `vendor/`
- `main.scss`

The abstracts need to be linked on every file of other folders, using `@use '../abstracts/index' as my;`, let's combine them on `abstracts/_index.scss`:

```scss
@use 'config';

@forward 'constants';
@forward 'font-stacks';
@forward 'functions' with (
  $my-base-url: config.$base-url,
  $my-breakpoints: config.$breakpoints,
  $my-font-weights: config.$font-weights,
  $my-gutter: config.$gutter,
  $my-palette: config.$palette,
);
@forward 'config';
@forward 'mixins';
```

The section 1 of the joining all file (which you insist on calling `main.scss` 😉 ) turns useless

```scss
@charset 'UTF-8';

// 1. Configuration and helpers
// @use 'abstracts/constants' as *;
// @use 'abstracts/stacks' as *;
// @use 'abstracts/functions' as *;
// @use 'abstracts/config' as *;
// @use 'abstracts/mixins' as *;

[...]
```

I'm considering to split the configurations in `_color-palettes.scss`, `_ratios-and-scales.scss`, `_general.scss`...

Too confusing?

**Comment by KittyGiraudel** [comment]
> Too confusing?

At this stage, yes. 😅 But I appreciate you taking the time to share your thoughts—thank you very much! I’ll try to make some time (and mind space) to come back to this issue soonish. :)

**Comment by robsonsobral** [comment]
I already found more things to improve. Soon I'll show you!

**Comment by Elrendio** [comment]
Amazing :star_struck: and completely agree!

I have a team specialized in front at my company and we're working on our guidelines. It could be the basis for Sass Guidelines (1.4). I'll create an issue by the end of the week 😊


PS: I'm very much impressed by your reaction time!

**Comment by TheElegantCoding** [comment]
I came here to see a real-world project now that sass is deprecated completely `@import` but it seems this is not implemented

**Comment by KittyGiraudel** [comment]
Yes, I haven’t used Sass since 2016 so I’m out of the loop and I wouldn’t feel comfortable writing an authoritative guide on this. I’m always happy with contributions though.
